ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/DefaultHostnameVerifier.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2015 - 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.typesafe.sslconfig.ssl
+
+import java.security.Principal
+import java.security.cert.{ Certificate, CertificateException, X509Certificate }
+
+import com.typesafe.sslconfig.util.LoggerFactory
+import javax.net.ssl.{ HostnameVerifier, SSLPeerUnverifiedException, SSLSession }
+import javax.security.auth.kerberos.KerberosPrincipal
+import sun.security.util.HostnameChecker
+
+@deprecated("DefaultHostnameVerifier has been deprecated and does nothing.  Please use the javax.net.debug system property.", "0.4.0")
+class DefaultHostnameVerifier(mkLogger: LoggerFactory) extends HostnameVerifier {
+  private val logger = mkLogger(getClass)
+
+  def hostnameChecker: HostnameChecker = {
+    logger.warn("DefaultHostnameVerifier has been deprecated and does nothing.  Please use the javax.net.debug system property.")
+    HostnameChecker.getInstance(HostnameChecker.TYPE_TLS)
+  }
+
+  def matchKerberos(hostname: String, principal: Principal) = {
+    logger.warn("DefaultHostnameVerifier has been deprecated and does nothing.  Please use the javax.net.debug system property.")
+    true
+  }
+
+  def isKerberos(principal: Principal): Boolean = {
+    logger.warn("DefaultHostnameVerifier has been deprecated and does nothing.  Please use the javax.net.debug system property.")
+    true
+  }
+
+  def verify(hostname: String, session: SSLSession): Boolean = {
+    logger.warn("DefaultHostnameVerifier has been deprecated and does nothing.  Please use the javax.net.debug system property.")
+    true
```

Comment:
this seems very dangerours, it did something before, now it does nothing. An upgrade can make you less secure. Should this not be false so that a user is forced to either use the Noop or provide a hostname verifier?

Reply:
Conversely being forced to do something is quite a breaking change, which is why I opted for true. What's better and what's worst? A false negative or a false positive? I'm guessing reasonable people will disagree.

If you feel strongly about this, I'm happy to flip the behaviour.